File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size() != t.size()) return 0;
        unordered_map<char, int> mp;
        for(char i : s){
            mp[i]++;
        }
        for(auto c : t){
            if(mp.find(c) == mp.end() || mp[c] == 0){
                return 0;
            }
            mp[c]--;
        }
        return 1;  
    }
};

File: Data Structures & Algorithms/reorder-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        """
        Do not return anything, modify head in-place instead.
        """
        #Find middle
        slow = head
        fast = head.next
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
        
        second = slow.next
        slow.next = None
        prev = None

        #Reverse second half
        while second:
            tmp = second.next
            second.next = prev
            prev = second
            second = tmp

        #Merge two halfs
        first, second = head, prev
        while second:
            target = first.next
            first.next = second
            first = second
            second = target
        
        

        

        
